### PR TITLE
Sexagesimal Output

### LINKED ISSUE #269: ability to output predict ra and dec in sexagesimal format with RA in hours and Dec in degrees
ability to output predict ra and dec in sexagesimal format with RA in hours and Dec in degrees - probably should be a command line flag with default in degrees for now

### PR DESCRIPTION
**Pull request by awilson110**
Fixes #269

Added function and command line flag to give users the option to have their output .csv file contain the coords in hours, minutes, seconds for RA and degrees, minutes, seconds for Dec. These are called ra_str_hms and dec_str_dms respectively. Before writing the .csv file the columns ra_str_hms and dec_str_dms are brought further to the front using pandas. Added known_sexagesimal.csv for unit testing that the conversion works correctly and handles edge cases well, and added holman_expected_predict_sg.csv to unit test that it outputs the columns correctly. 

## Review Checklist for Source Code Changes

- [x] Does pip install still work?
- [x] Have you written a unit test for any new functions?
- [x] Do all the units tests run successfully?
- [x] Does Layup run successfully on a test set of input files/databases?
- [x] Have you used black on the files you have updated to confirm python programming style guide enforcement?

### CODE REVIEW

File: src/layup/predict.py [hunk 1]
```diff
@@ -55,6 +55,40 @@ def _get_result_dtypes(primary_id_column_name: str):
...
+
+    Returns
+    -------
+    input array with ra and dec in sexagesimal appended, called ra_str_hms and dec_str_dms respectively.
+    """
+    ra_deg = (data["ra_deg"] / 15) % 24.0  # Ensuring ra is within 24 hours/360 degrees
+    dec_deg = data["dec_deg"]
+    ra_decimal = ra_deg % 1.0  # Take decimal portion to find arcminutes
+    dec_decimal = np.abs(dec_deg) % 1
+    ra_m = ra_decimal * 60
```
Comment (mschwamb) [inline src/layup/predict.py]: make these ints give they are ints in the string

File: src/layup/predict.py [hunk 2]
```diff
@@ -55,6 +55,40 @@ def _get_result_dtypes(primary_id_column_name: str):
...
+    dec_m = dec_decimal * 60
+    ra_decimal = ra_m % 1  # Take decimal portion again for arcseconds
+    dec_decimal = dec_m % 1
+    ra = np.empty(len(ra_deg), dtype="<U16")
+    dec = np.empty(len(ra_deg), dtype="<U16")
+    for i in range(len(ra_deg)):
```
Comment (mschwamb) [inline src/layup/predict.py]: add an empty new line for read ability before line 80

File: src/layup/predict.py [hunk 3]
```diff
@@ -55,6 +55,40 @@ def _get_result_dtypes(primary_id_column_name: str):
...
+    """
+    ra_deg = (data["ra_deg"] / 15) % 24.0  # Ensuring ra is within 24 hours/360 degrees
+    dec_deg = data["dec_deg"]
+    ra_decimal = ra_deg % 1.0  # Take decimal portion to find arcminutes
+    dec_decimal = np.abs(dec_deg) % 1
+    ra_m = ra_decimal * 60
+    dec_m = dec_decimal * 60
+    ra_decimal = ra_m % 1  # Take decimal portion again for arcseconds
+    dec_decimal = dec_m % 1
+    ra = np.empty(len(ra_deg), dtype="<U16")
```
Comment (mschwamb) [inline src/layup/predict.py]: add a blank line for readability before line 78

File: src/layup_cmdline/predict.py [hunk 4]
```diff
@@ -175,6 +175,16 @@ def main():
         required=False,
     )
 
+    optional.add_argument(
+        "-us",
+        "--units-sexagesimal",
+        help="Option for output units of RA and Dec to be given in sexagesimal. False will use degrees.",
+        dest="units",
```
Comment (mschwamb) [inline src/layup_cmdline/predict.py]: because you have dest units - it's looking for the user to put some after the -us on the command line. There isn't anything for the user to specify. You want this to be like force flag. The reason why is it it defaults to false I don't need to raise the flag unless I want it to be true

File: src/layup/utilities/file_io/file_output.py [hunk 5]
```diff
@@ -14,8 +14,19 @@ def write_csv(data, filepath):
         The data to write to the file.
     filepath : str
         The path to the file to write.
+    move_columns : dict, optional
+        Dict of any column names that need moved, paired with their new position.
     """
     df = pd.DataFrame(data)
+
+    if move_columns != {}:
+        column_names = list(df.columns.values)
```
Comment (mschwamb) [inline src/layup/utilities/file_io/file_output.py]: what happens if the user sent a column name that doesn't exist in the data array? Need to check and error on that. you might be able to do a try and except

https://www.w3schools.com/python/python_try_except.asp

**Review by mschwamb — CHANGES_REQUESTED** [review]
Getting closer

File: src/layup/utilities/file_io/file_output.py [hunk 6]
```diff
@@ -5,7 +5,7 @@
 import pandas as pd
 
 
-def write_csv(data, filepath):
+def write_csv(data, filepath, move_columns={}):
```
Comment (mschwamb) [inline src/layup/utilities/file_io/file_output.py]: Also I think it's better to not have an empty array as the default value. Better to use None instead of {} https://www.codiga.io/blog/python-avoid-empty-array-parameter/

File: src/layup/utilities/file_io/file_output.py [hunk 7]
```diff
@@ -14,8 +14,22 @@ def write_csv(data, filepath):
...
     df = pd.DataFrame(data)
+
+    if move_columns != {}:
+        column_names = list(df.columns.values)
+        for col in move_columns.keys():
+            try:
+                column_names.pop(column_names.index((col)))
+                column_names.insert(move_columns[col], col)
+            except:
+                raise ValueError(f"column {col} not found in df.columns.values.")
```
Comment (mschwamb) [inline src/layup/utilities/file_io/file_output.py]: Have you tested that this works by sending the code the incorrect thing through? It would be good to add this to the unit tests. https://stackoverflow.com/questions/129507/how-do-you-test-that-a-python-function-throws-an-exception

**Review by mschwamb — CHANGES_REQUESTED** [review]
I think this should be the last set of requested changes.

File: src/layup/utilities/file_io/file_output.py (hunk 7 above)
Comment (awilson110) [reply src/layup/utilities/file_io/file_output.py]: I added a new unit test to test the column moving of write_csv, and one thing it tests is if returns an error for nonexistent columns
